api/app/controllers/webhook.controller.ts

```diff
@@ -154,3 +156,81 @@ export async function create(req: Request, res: Response) {
     }
   }
 }
+
+export async function handlePushEvent(req: Request, res: Response) {
+  if (!verify_signature(req)) {
+    return res.status(401).json("Unauthorized");
+  }
+
+  const pushEvent = req.body;
+  if (!pushEvent) {
+    return res.status(500).json({
+      message: "No push event found in the request body.",
+    });
+  }
+
+  const commits = pushEvent.commits;
+  if (!commits) {
+    return res.status(500).json({
+      message: "No commits found in the request body.",
+    });
+  }
+
+  let responseStatus = 200;
+  let responseMessage = "success";
+
+  try {
+    for (const commit of commits) {
+      const changedFiles = [...commit.added, ...commit.modified];
+      for (const file of changedFiles) {
+        const rawUrl = `https://raw.githubusercontent.com/${pushEvent.repository.full_name}/master/${file}`;
+        const response: AxiosResponse<TSTBTCAttributes> = await axios.get(rawUrl);
+        const mdContent = response.data;
+        const jsonContent = parseMdToJSON(mdContent);
+
+        const transcriptHash = generateUniqueHash(jsonContent);
+        const totalWords = getTotalWords(jsonContent.body);
+        const content = jsonContent;
+        const originalContent = jsonContent;
+
+        const existingTranscript = await Transcript.findOne({
+          where: { transcriptHash: transcriptHash },
+        });
+
+        if (existingTranscript) {
+          responseStatus = 409;
+          responseMessage = "transcript already exists";
+          break;
+        }
+
+        if (jsonContent.transcript_by.includes("TSTBTC")) {
+          await Transcript.create({
+            transcriptUrl: rawUrl,
+            transcriptHash: transcriptHash,
+            contentTotalWords: totalWords,
+            content: content,
+            originalContent: originalContent,
+            status: TranscriptStatus.queued,
+          });
+          break;
+        } else {
+          responseStatus = 404;
+          responseMessage =
+            "Transcript not from TSTBTC - did not queue transcript";
+        }
+      }
+
+      if (responseStatus !== 200) {
+        break;
+      }
+    }
+  } catch (error) {
```

Comment:
We need proper error handling here. In the case of an error, we should probably have an alert system to notify us if there is an unsuccessful save. If not, the transcript will be on github but our backend won't have any knowledge of it

Reply:
This already captures the error you highlighted, if a transcript gets to GitHub from tstbtc, we are optimistic the push event will trigger(except it does not), if it does not trigger our endpoint does not get anything(which means we can't process anything further since our endpoint won't trigger). What we can do here is to include our logger to report errors in observability(is that what you mean?).

Reply:
The call clarifies the issue. We can set up a node-mailer function using Gmail as a transporter to notify such events.

Reply:
We can track the response through https://github.com/bitcointranscripts/bitcointranscripts/settings/hooks/415799675?tab=deliveries

I think that this is enough for now. When I push a new transcript, I can immediately see if it was added successfully.

Reply:
@kouloumos tracking the deliveries is very crude and won't suffice for the new implementation. Its very easy for any of the helper functions to fail and throw an error that triggers a 500 delivery to github but we have no idea that it happened except we manually check github.
In end we have a diff between our db and the repo.